tools/generators/pbxproject_targets/src/Generator/IdentifyTargets.swift

```diff
@@ -0,0 +1,90 @@
+import Foundation
+import GeneratorCommon
+import PBXProj
+
+struct IdentifiedTarget: Equatable {
+    let consolidationMapOutputPath: URL
+    let key: ConsolidatedTarget.Key
+    let identifier: Identifiers.Targets.Identifier
+    let dependencies: [TargetID]
+}
+
+extension Generator {
+    struct IdentifyTargets {
+        private let consolidateTargets: ConsolidateTargets
+        private let disambiguateTargets: DisambiguateTargets
+        private let innerIdentifyTargets: InnerIdentifyTargets
+
+        private let callable: Callable
+
+        /// - Parameters:
+        ///   - callable: The function that will be called in
+        ///     `callAsFunction()`.
+        init(
+            consolidateTargets: ConsolidateTargets,
+            disambiguateTargets: DisambiguateTargets,
+            innerIdentifyTargets: InnerIdentifyTargets,
+            callable: @escaping Callable = Self.defaultCallable
+        ) {
+            self.consolidateTargets = consolidateTargets
+            self.disambiguateTargets = disambiguateTargets
+            self.innerIdentifyTargets = innerIdentifyTargets
+
+            self.callable = callable
+        }
+        
+        /// Identifies all the targets defined by the consolidation map
+        /// arguments.
+        func callAsFunction(
+            consolidationMapArguments: [ConsolidationMapArguments],
+            logger: Logger
+        ) throws -> [IdentifiedTarget] {
+            return try callable(
+                /*consolidationMapArguments:*/ consolidationMapArguments,
+                /*logger:*/ logger,
+                /*consolidateTargets:*/ consolidateTargets,
+                /*disambiguateTargets:*/ disambiguateTargets,
+                /*innerIdentifyTargets:*/ innerIdentifyTargets
+            )
+        }
+    }
+}
+
+// MARK: - IdentifyTargets.Callable
+
+extension Generator.IdentifyTargets {
+    public typealias Callable = (
+        _ consolidationMapArguments: [ConsolidationMapArguments],
+        _ logger: Logger,
+        _ consolidateTargets: Generator.ConsolidateTargets,
+        _ disambiguateTargets: Generator.DisambiguateTargets,
+        _ innerIdentifyTargets: Generator.InnerIdentifyTargets
+    ) throws -> [IdentifiedTarget]
+
+    // TODO: Add test
```

Comment:
Last test technically needed, but I'm too tired to do it, and it's low value imo.